File: snippets/fsharp/System/Double/NaN/nan1.fs
module nan1

open System

// <Snippet1>
let zero = 0.0
printfn $"{zero} / {zero} = {zero / zero}"
// The example displays the following output:
//         0 / 0 = NaN
// </Snippet1>

// <Snippet2>
let nan1 = Double.NaN

printfn $"{3} + {nan1} = {3. + nan1}"
printfn $"abs({nan1}) = {abs nan1}"
// The example displays the following output:
//       3 + NaN = NaN
//       abs NaN = NaN
// </Snippet2>

// <Snippet3>
let result = Double.NaN
printfn $"{result} = Double.Nan: {result = Double.NaN}"
// The example displays the following output:
//         NaN = Double.Nan: False
// </Snippet3>

File: snippets/csharp/System/Single/NaN/nan1.cs
using System;

public class Example
{
    public static void Main()
    {
        // <Snippet1>
        float zero = 0.0f;
        Console.WriteLine($"{zero} / {zero} = {zero / zero}");
        // The example displays the following output:
        //         0 / 0 = NaN
        // </Snippet1>

        // <Snippet2>
        float nan1 = float.NaN;

        Console.WriteLine($"{3} + {nan1} = {3 + nan1}");
        Console.WriteLine($"Abs({nan1}) = {Math.Abs(nan1)}");
        // The example displays the following output:
        //       3 + NaN = NaN
        //       Abs(NaN) = NaN
        // </Snippet2>
        Console.WriteLine();

        // <Snippet3>
        float result = float.NaN;
        Console.WriteLine($"{result} = Single.NaN: {result == float.NaN}");
        // The example displays the following output:
        //         NaN = Single.Nan: False
        // </Snippet3>
    }
}
